core/backend.go

```diff
@@ -52,11 +52,12 @@ type Backend interface {
 	MessageConstructor
 	Verifier
 
-	// BuildProposal builds a new block proposal
-	BuildProposal(view *proto.View) []byte
+	// BuildProposal builds a new ethereum block
+	BuildEthereumBlock(height uint64) []byte
 
 	// InsertBlock inserts a proposal with the specified committed seals
-	InsertBlock(proposal []byte, committedSeals []*messages.CommittedSeal)
+	// the reason why we are including round here is because a single committedSeal has signed the tuple of (EB, r)
+	InsertBlock(ethereumBlock []byte, round uint64, committedSeals []*messages.CommittedSeal)
```

Comment:
Could we simplify this signature, by providing just `ProposedBlock` instance and would it be correct thing to do?

Hopefully `state.view.round` and `state.proposalMessage.ProposedBlock.round` matches if `proposalMessage` is set? Also hope we should provide round from the `proposalMessage` (if there is no `proposalMessage` set, then it is all the same if we provide 0 or some arbitrary number greater than 0 for the round). This is at least my understanding. Am I wrong?

```suggestion
	InsertProposal(proposal *proto.ProposedBlock, committedSeals []*messages.CommittedSeal)
```

Reply:
Fixed 7b4bec9f521ee049311a7ae364d69a96cd66c655